sds_in_a_box/sds_in_a_box_stack.py

```diff
@@ -72,27 +83,122 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+                                    account_recovery=cognito.AccountRecovery.EMAIL_ONLY,
+                                    auto_verify=cognito.AutoVerifiedAttrs(email=True),
+                                    standard_attributes=cognito.
+                                    StandardAttributes(email=cognito.StandardAttribute(required=True)),
+                                    sign_in_aliases=cognito.SignInAliases(username=False, email=True),
+                                    removal_policy=cdk.RemovalPolicy.DESTROY
+                                    )
+
+        # Add a client sign in for the userpool
+        command_line_client = cognito.UserPoolClient(user_pool=userpool, scope=self, id='sds-command-line',
+                                                     user_pool_client_name= f"sdscommandline-{SDS_ID}",
+                                                     id_token_validity=cdk.Duration.minutes(60),
+                                                     access_token_validity=cdk.Duration.minutes(60),
+                                                     refresh_token_validity=cdk.Duration.minutes(60),
+                                                     auth_flows=cognito.AuthFlow(admin_user_password=True,
+                                                                                 user_password=True,
+                                                                                 user_srp=True,
+                                                                                 custom=True),
+                                                     prevent_user_existence_errors=True)
+        
+        # Add a random unique domain name where users can sign up / reset passwords
+        # Users will be able to reset their passwords at https://sds-login-{SDS_ID}.auth.us-west-2.amazoncognito.com/login?client_id={}&redirect_uri=https://example.com&response_type=code
+        userpooldomain = userpool.add_domain(id="TeamLoginCognitoDomain",
+                                             cognito_domain=cognito.CognitoDomainOptions(domain_prefix=f"sds-login-{SDS_ID}"))
+
+        # Add a lambda function that will trigger whenever an email is sent to the user (see the lambda section above)
+        userpool.add_trigger(cognito.UserPoolOperation.CUSTOM_MESSAGE, signup_lambda)
+
+        # Create an initial user of the API
+        initial_user = cognito.CfnUserPoolUser(self, "MyCfnUserPoolUser",
+                                               user_pool_id=userpool.user_pool_id,
+                                               desired_delivery_mediums=["EMAIL"],
+                                               force_alias_creation=False,
+                                               user_attributes=[cognito.CfnUserPoolUser.AttributeTypeProperty(
+                                                  name="email",
+                                                  value="harter@lasp.colorado.edu"
```

Comment:
remove individual credentials here.

If you want this set automatically I suppose you could add this in as an argument into the stack itself so everyone could pass in what they want.